derp/derphttp/mesh_client.go

```diff
@@ -121,6 +125,9 @@ func (c *Client) RunWatchConnectionLoop(ctx context.Context, ignoreServerKey key
 		// Make sure we're connected before calling s.ServerPublicKey.
 		_, _, err := c.connect(ctx, "RunWatchConnectionLoop")
 		if err != nil {
+			if notifyError != nil {
+				notifyError(err)
+			}
```

Comment:
:thinking: Blocking concern: I think we should we hoist line 134 above this, so that the error is logged before the callback has a chance to panic: 
https://github.com/tailscale/tailscale/blob/136e0834696e8c9c8e7ca1b34b04b4272351b287/derp/derphttp/mesh_client.go#L127-L137

```suggestion
			logf("mesh connect: %v", err)
			if notifyError != nil {
				notifyError(err)
			}
```

This is done properly below, on line 150:
https://github.com/tailscale/tailscale/blob/136e0834696e8c9c8e7ca1b34b04b4272351b287/derp/derphttp/mesh_client.go#L148-L156